src/NHibernate/Hql/Ast/ANTLR/ASTQueryTranslatorFactory.cs

```diff
@@ -33,8 +34,17 @@ public IQueryTranslator[] CreateQueryTranslators(IQueryExpression queryExpressio
 
 			var translators = polymorphicParsers
 				.ToArray(hql => queryExpression is NhLinqExpression linqExpression
```

Comment:
Don't you need to change this cast to `ILinqQueryExpression` too?

Reply:
I guess you are right. I'll change that, but have to publish the `GetNamedParameterTypes()` via the `ILinqQueryExpression` of course too.